utils/prompt_generation.py

```diff
@@ -0,0 +1,424 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# SPDX-FileCopyrightText: © 2024 Tenstorrent AI ULC
+
+import os
+from pathlib import Path
+import logging
+import argparse
+import json
+from datetime import date
+
+import torch
+from jinja2 import Template
+from datasets import load_dataset
+from transformers import AutoTokenizer
+
+
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+# set torch seed for reproducibility
+torch.manual_seed(42)
+
+
+def load_alpaca_eval_dataset_samples(num_prompts):
+    # Load alpaca_eval dataset with specified number of samples
+    alpaca_ds = load_dataset(
+        "tatsu-lab/alpaca_eval",
+        "alpaca_eval",
+        split=f"eval[:{num_prompts}]",
+        trust_remote_code=True,
+    )
+    return alpaca_ds["instruction"]
+
+
+def tokenize_encode(prompt, tokenizer, max_length, tokenizer_model):
+    if tokenizer_model == "meta-llama/Llama-3.1-70B-Instruct":
+        return tokenizer.encode(
+            prompt, add_special_tokens=False, truncation=True, max_length=max_length
+        )
+    else:
+        raise ValueError(f"Unsupported tokenizer model: '{tokenizer_model}'.")
+
+
+def tokenize_decode(encoded_prompt, tokenizer, tokenizer_model):
+    if tokenizer_model == "meta-llama/Llama-3.1-70B-Instruct":
+        return tokenizer.decode(encoded_prompt)
+    else:
+        raise ValueError(f"Unsupported tokenizer model: '{tokenizer_model}'.")
+
+
+# Define a function to generate random prompts using a model's vocabulary
+def generate_random_prompts(
+    num_prompts: int,
+    max_length: int,
+    input_seq_len: int,
+    distribution: str,
+    tokenizer_model=None,
+):
+    """
+    Generate random prompts using the model's vocabulary.
```

Comment:
Love docstrings. We should align on a standard for consistency. Should they be used for all Classes, Methods, and Functions.

Reply:
Agreed, we can use a prompt template for generation draft doc strings that meet that standard too!